src/services/refactors/extractType.ts

```diff
@@ -205,21 +228,33 @@ function flattenTypeLiteralNodeReference(checker: TypeChecker, node: TypeNode |
         }
         return result;
     }
-    else if (isParenthesizedTypeNode(node)) {
-        return flattenTypeLiteralNodeReference(checker, node.type);
+    else if (isParenthesizedTypeNode(selection)) {
+        return flattenTypeLiteralNodeReference(checker, selection.type);
     }
-    else if (isTypeLiteralNode(node)) {
-        return node.members;
+    else if (isTypeLiteralNode(selection)) {
+        return selection.members;
     }
     return undefined;
 }
 
-function rangeContainsSkipTrivia(r1: TextRange, node: Node, file: SourceFile): boolean {
-    return rangeContainsStartEnd(r1, skipTrivia(file.text, node.pos), node.end);
+function rangeContainsSkipTrivia(r1: TextRange | TextRange[], node: Node | Node[], file: SourceFile): boolean {
+    const r1Range = arrayToNode(r1);
+    const nodeRange = arrayToNode(node);
+    return rangeContainsStartEnd(r1Range, skipTrivia(file.text, nodeRange.pos), nodeRange.end);
+
+    function arrayToNode(rangeOrArray: TextRange | TextRange[]): TextRange {
+        return isArray(rangeOrArray) ? { pos: rangeOrArray[0].pos, end: rangeOrArray[rangeOrArray.length - 1].end } : rangeOrArray;
+    }

```

Comment:
Are you sure these changes ended up being needed? I don’t see any change to the caller. It looks like it’s always just passing a `TextRange`.

Reply:
In `visitor` in `collectTypeParameters`, this function was called with a reference to an un-narrowed `selection` as `r1`

Reply:
After some more thought, this code will only ever be converting `selection` into a `TextRange`, it might be better to do that once, and pass that range into `rangeContainsSkipTrivia`, as I don't see why `node` needs to be a `Node`